mm2src/coins/eth.rs

```diff
@@ -1468,48 +1435,75 @@ impl WatcherOps for EthCoin {
                 )));
             }
 
+            if tx_from_rpc.to != Some(expected_swap_contract_address) {
+                if tx_from_rpc.to == fallback_swap_contract {
+                    expected_swap_contract_address = fallback_swap_contract.ok_or_else(|| {
+                        ValidatePaymentError::InternalError("Fallback swap contract address not found".to_string())
+                    })?;
+                } else {
+                    return MmError::err(ValidatePaymentError::WrongPaymentTx(format!(
+                        "Payment tx {:?} was sent to wrong address, expected {:?}",
+                        tx_from_rpc, expected_swap_contract_address,
+                    )));
+                }
+            }
```

Comment:
I think it can be simplified a bit. Please consider these changes
```rust
if tx_from_rpc.to != self.swap_contract_address && Some(tx_from_rpc.to) != fallback_swap {
  return MmError::err(ValidatePaymentError::WrongPaymentTx(format!(
                        "Payment tx {tx_from_rpc:?} was sent to wrong address, expected either {:?} or the fallback {fallback_swap_contract:?}", self.swap_contract_address,
                    )));
}
let expected_swap_contract = tx_from_rpc.to;
```

Reply:
Nice, yes this is better